pallets/file-system/src/lib.rs

```diff
@@ -528,10 +517,16 @@ pub mod pallet {
             size: StorageData<T>,
             peer_ids: PeerIds<T>,
         },
-        /// Notifies that a MSP has responded to storage request(s).
-        MspRespondedToStorageRequests {
-            results: MspRespondStorageRequestsResult<T>,
-        },
+        /// Notifies that a Main Storage Provider (MSP) has accepted a storage request for a specific file key.
+        ///
+        /// This event is emitted when an MSP agrees to store a file, but the storage request
+        /// is not yet fully fulfilled (i.e., the required number of Backup Storage Providers
+        /// have not yet confirmed storage).
+        ///
+        /// # Note
+        /// This event is not emitted when the storage request is immediately fulfilled upon
+        /// MSP acceptance. In such cases, a [`StorageRequestFulfilled`] event is emitted instead.
```

Comment:
Why not emitting both?

Reply:
Only to reduce the onchain size, since it's not adding any additional information in that case (fulfilled request is a superset).

Reply:
I mean, it's not that big of an optimisation, but I get it 👌🏼